lighthouse-core/gather/driver.js

```diff
@@ -741,6 +792,14 @@ class Driver {
     /** @type {NodeJS.Timer|undefined} */
     let maxTimeoutHandle;
 
+    // Noop if offline or not https
+    // https: => https
+    const protocol = new URL(this._monitoredUrl || '').protocol.replace(/:$/, '');
```

Comment:
It's possible we can make this approach work, but I don't think it will work like this (correct me if I'm missing something).

`_monitoredUrl` is updated as the page loads to account for redirects, so we would have to recheck every time it updates whether or not we should be checking for security state changes. So I believe a site that redirects http -> https won't get a check since the value will initially be `!isSecureProtocol`?

Reply:
Good call, Brendan.

Example URL that redirects to insecure (but isn't caught currently): http://i.paul.irish  
_(as soon as DNS propagates, http://expiredredirect.paul.irish will do the same)_